server/lib/validation/languages/th.js

```diff
@@ -56,43 +56,43 @@ const INVALIDATIONS = [{
   // - Tone marks
   // - Phinthu, Thanthakhat, Nikhahit, Yamakkan
   /* eslint-disable-next-line no-misleading-character-class */
-  regex: /(^|\s+)[\u0E30\u0E32\u0E33\u0E45\u0E31\u0E34\u0E35\u0E36\u0E37\u0E4D\u0E47\u0E38\u0E39\u0E48\u0E49\u0E4A\u0E4B\u0E3A\u0E4C\u0E4D\u0E4E]/,
+  regex: /(^|\s)[\u0E30\u0E32\u0E33\u0E45\u0E31\u0E34\u0E35\u0E36\u0E37\u0E4D\u0E47\u0E38\u0E39\u0E48\u0E49\u0E4A\u0E4B\u0E3A\u0E4C\u0E4D\u0E4E]/,
   error: 'Word should not start with unexpected characters',
 }, {
   // These Thai chars cannot end the word:
   // - Lead vowels
-  regex: /[\u0E40\u0E41\u0E42\u0E43\u0E44](\s+|$)/,
+  regex: /[\u0E40\u0E41\u0E42\u0E43\u0E44](\s|$)/,
   error: 'Word should not end with leading vowels',
 }, {
   // Any words consisting of letters with a period
   // inbetween are considered abbreviations or acronyms.
   // Abbreviations in Latin chars are disallowed by previous rules already.
-  regex: /[ก-ฮ]+\.([ก-ฮ]+\.)+/,
+  regex: /[ก-ฮ]\.[ก-ฮ]{1,5}\./,
```

Comment:
Can you elaborate on this? Are there no abbreviations that have a longer second part than 5 characters?

Reply:
The longest known sequence of continuous letters for any abbreviation part is 4.
- As in `รง.ซย.กรซท.ศซส.สพ.ทบ.` https://hilight.kapook.com/view/139424 

Text like `กขคงจฉช.กขคงจ.กขคงจฉ.กขคงจฉช.กขคงจฉ.` would not considered an abbreviation but more like a running text with `.` as delimiter.

This is heuristic anyway. We can have `/[ก-ฮ]\.[ก-ฮ]+\./` as well for simplicity.

Reply:
As long as the rule doesn't allow sentences that are hard to read for contributors all should be fine :) As I have no idea about Thai, I'll leave that up to you.

Reply:
I committed a new code with `/[ก-ฮ]\.[ก-ฮ]+\./` for simplicity.